### PR TITLE
Widget and Registration Forms [Concludes #92]

### PR DESCRIPTION
**Pull request by nodatall**
> Success consists of going from form to form without loss of enthusiasm. - Winston Churchill

### CODE REVIEW

**Comment by ameliavoncat** [comment]
Forms have also been reduxed.

File: src/components/forms/Form.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React, { Component } from 'react'
+
+export default class Form extends Component {
+
```
Comment (jason00111) [inline src/components/forms/Form.js:4]: remove extra line

File: src/components/forms/RegistrationForm.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  getChildContext() {
+    return {
+      onChange: this.onChange.bind( this ),
+      onChecked: this.onChecked.bind( this )
+    }
+  }
+
+  render() {
+    return (
+      <form action="http://localhost:5000/api/v1/users/register" method="post">
```
Comment (jason00111) [inline src/components/forms/RegistrationForm.js]: Use form component here

File: src/components/forms/form_components/Checkbox.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  render() {
+    const { name, defaultValue } = this.props
+    const { onChecked } = this.context
+
+    return (
+      <Checkbox
+        fieldName={ name }
+        defaultValue={ defaultValue }
+        onChecked={ onChecked }
+      />
```
Comment (jason00111) [inline src/components/forms/form_components/Checkbox.js]: move this up to line 16

File: src/reducers/forms/index.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { FORM_FIELD_UPDATE, FORM_RESET } from '../../actions/form/constants'
+
+const initialState = {
+  values: {}
+}
+
+const formFieldUpdate = ( state, action ) => {
+  return Object.assign({}, state, {
+    values: Object.assign({}, state.values, {
+      [action.name]: action.value
```
Comment (jason00111) [inline src/reducers/forms/index.js]: add spaces inside []

File: src/reducers/forms/index.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    case FORM_RESET:
+      return initialState
+    case FORM_FIELD_UPDATE:
+      return formFieldUpdate( state, action )
+    default:
+      return state
+  }
+}
+
+export default formReducer
```
Comment (jason00111) [inline src/reducers/forms/index.js]: This file looks like a duplicate of form/index.js.  Is this intentional?

File: src/reducers/form/formReducer.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+export const formFieldUpdate = ( state, action ) => {
+  const { name, value } = action
+
+  const newFieldValue = Object.assign( {}, state )
+  newFieldValue.data = {
+    [name]: value
+  }
+
+  return newFieldValue
+}
```
Comment (jason00111) [inline src/reducers/form/formReducer.js:7]: This file isn't being used... what's up with that?

File: src/components/forms/form_components/WidgetForm/WidgetPreview.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+import "../../../../styles/forms/form.css"
+
+export default class WidgetPreview extends Component {
+  render() {
+    const { description, image, title } = this.props
+
+    return (
+      <div className="WidgetPreview">
+        <img alt={title} src={image} />
+        <div>{description}</div>
```
Comment (jason00111) [inline src/components/forms/form_components/WidgetForm/WidgetPreview.js:11]: run `yarn lint:fix`

**Review by jason00111 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**